go.mod

```diff
@@ -1,26 +1,42 @@
 module github.com/warrensbox/terraform-switcher
 
-go 1.13
+go 1.18
```

Comment:
Local build went fine apart from this strange error at the very beginning:
```
> make release

fatal: unable to connect to github.com:
github.com[0: 140.82.121.3]: errno=Connection timed out

go build -v -ldflags "-X main.version=" -o tfswitch github.com/warrensbox/terraform-switcher
[…]
github.com/warrensbox/terraform-switcher
GOOS=darwin go build -ldflags "-X main.version=" -o tfswitch--darwin-amd64 github.com/warrensbox/terraform-switcher
GOOS=linux go build -ldflags "-X main.version=" -o tfswitch--linux-amd64 github.com/warrensbox/terraform-switcher
```
The error itself comes from the `Makefile`'s line:
```
VER := $(shell git ls-remote --tags git://github.com/warrensbox/terraform-switcher | awk '{print $$2}'| awk -F"/" '{print $$3}' | sort -n -t. -k1,1 -k2,2 -k3,3 | tail -n 2 | head -n1)
```
Which I believe should had either `git@github.com:warrensbox/terraform-switcher.git` URL (preferred) or `https://github.com/warrensbox/terraform-switcher.git` (not preferred since recently they disabled remote anonymous access to repos by default).

Also that `Makefile`'s line can be a bit optimized (at the cost of readability maybe 🤔) like below:
```
git ls-remote --tags git@github.com:warrensbox/terraform-switcher.git | awk '{if ($2 ~ "\\^\\{\\}$") next; print vers[split($2,vers,"\\/")]}' | sort -n -t. -k
1,1 -k2,2 -k3,3 | tail -1
```
Feel free to incorporate this into this PR is you think it may make sense.

Reply:
Most probably it is not related, but all of `--show-latest*` cmdline parameters bring up the selection menu for me instead of showing respective versions 😕

Reply:
Thanks will update this as well.
on the `show-latest*` flags I'm not seeing the same behavior when built of this branch.
```
➜  terraform-switcher git:(techdebt/go-upgrade) $  ./tfswitch--darwin-amd64 -S 1.1

Matched version: 1.1.9
1.1.9

➜  terraform-switcher git:(techdebt/go-upgrade) $ ./tfswitch--darwin-amd64 -P 1.1

1.1.0-rc1
```

Reply:
Oh, my bad, I copy-paste'd this from command line and missed to do the proper escaping. Here's the good line:
```
VER := $(shell git ls-remote --tags git@github.com:warrensbox/terraform-switcher.git | awk '{if ($$2 ~ "\\^\\{\\}$$") next; print vers[split($$2,vers,"\\/")]}' | sort -n -t. -k1,1 -k2,2 -k3,3 | tail -1)
```